tools/runner/update_manifest.py

```diff
@@ -7,21 +7,31 @@
 
 root = localpaths.repo_root
 
-from manifest import manifest, update
+from manifest import manifest
 
 def main(request, response):
     path = os.path.join(root, "MANIFEST.json")
 
-    manifest_file = None
-    try:
-        manifest_file = manifest.load(root, path)
-    except manifest.ManifestVersionMismatch:
-        pass
-    if manifest_file is None:
-        manifest_file = manifest.Manifest("/")
-
-    update.update(root, manifest_file)
-
-    manifest.write(manifest_file, path)
-
-    return [("Content-Type", "application/json")], json.dumps({"url": "/MANIFEST.json"})
+    # TODO make this download when required
+    manifest_file = manifest.load_and_update(root, path, "/", parallel=False)
+
+    data = {"items": {},
+            "url_base": "/"}
+    for item_type, path, tests in manifest_file.itertypes():
+        if item_type not in ["testharness", "reftest", "manual"]:
```

Comment:
Could this not just pass `["testharness", "reftest", "manual"]` to `manifest_file.itertypes()`? (I may be missing something)

Reply:
Good point.